### ISSUE TITLE
The example gets stuck

### ISSUE THREAD

**Issue by georgepachitariu** [criteo/tf-yarn#43]
Hi, 
thanks for writing this library.
I'm trying to run [this](https://github.com/criteo/tf-yarn/blob/master/examples/linear_classifier_example.py) example for the first time, but the job gets stuck. The chief log is below.

I also tried the other examples but none of them work and I'm not sure how to debug this. Can you please give me some pointers (baby steps) on what I should try?

This is the Chief log:
+ pyenv/bin/python -m tf_yarn.tasks._independent_workers_task
2020-08-21 12:49:47,712:INFO:tf_yarn._task_commons: Python 3.6.10 | packaged by conda-forge | (default, Apr 24 2020, 16:44:11) 
[GCC 7.3.0]
2020-08-21 12:49:47,712:INFO:tf_yarn._task_commons: Skein 0.8.0
2020-08-21 12:49:47,712:INFO:tf_yarn._task_commons: TensorFlow unknown 2.0.0
2020-08-21 12:50:30.067074: I tensorflow/core/platform/cpu_feature_guard.cc:145] This TensorFlow binary is optimized with Intel(R) MKL-DNN to use the following CPU instructions in performance critical operations:  SSE4.1 SSE4.2 AVX AVX2 FMA
To enable them in non-MKL-DNN operations, rebuild TensorFlow with the appropriate compiler flags.
2020-08-21 12:50:30.080840: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2100155000 Hz
2020-08-21 12:50:30.082179: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x561beab2a910 executing computations on platform Host. Devices:
2020-08-21 12:50:30.082222: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Host, Default Version
2020-08-21 12:50:30.143241: I tensorflow/core/common_runtime/process_util.cc:115] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
2020-08-21 12:50:30,145:INFO:tensorflow: TF_CONFIG environment variable: {'cluster': {'chief': ['10.1.18.46:34146'], 'ps': ['10.1.18.6:41640']}, 'environment': 'google', 'task': {'type': 'chief', 'index': 0}}
2020-08-21 12:50:30,146:INFO:tensorflow: Using default config.
2020-08-21 12:50:30,146:INFO:tensorflow: Using config: {'_model_dir': 'hdfs://nameservice1/user/gpachitariu/tf', '_tf_random_seed': None, '_save_summary_steps': 100, '_save_checkpoints_steps': None, '_save_checkpoints_secs': 600, '_session_config': device_filters: "/job:ps"
device_filters: "/job:chief"
allow_soft_placement: true
graph_options {
  rewrite_options {
    meta_optimizer_iterations: ONE
  }
}
, '_keep_checkpoint_max': 5, '_keep_checkpoint_every_n_hours': 10000, '_log_step_count_steps': 100, '_train_distribute': None, '_device_fn': None, '_protocol': None, '_eval_distribute': None, '_experimental_distribute': None, '_experimental_max_worker_delay_secs': None, '_session_creation_timeout_secs': 7200, '_service': None, '_cluster_spec': <tensorflow.python.training.server_lib.ClusterSpec object at 0x7f830606e780>, '_task_type': 'chief', '_task_id': 0, '_evaluation_master': '', '_master': 'grpc://10.1.18.46:34146', '_num_ps_replicas': 1, '_num_worker_replicas': 1, '_global_id_in_cluster': 0, '_is_chief': True}
2020-08-21 12:50:30,154:INFO:__main__: Starting server chief:0
E0821 12:50:30.157408065   87747 socket_utils_common_posix.cc:198] check for SO_REUSEPORT: {"created":"@1598014230.157387580","description":"SO_REUSEPORT unavailable on compiling system","file":"external/grpc/src/core/lib/iomgr/socket_utils_common_posix.cc","file_line":166}
2020-08-21 12:50:30.157683: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:258] Initialize GrpcChannelCache for job chief -> {0 -> localhost:34146}
2020-08-21 12:50:30.157704: I tensorflow/core/distributed_runtime/rpc/grpc_channel.cc:258] Initialize GrpcChannelCache for job ps -> {0 -> 10.1.18.6:41640}
2020-08-21 12:50:30.159426: I tensorflow/core/distributed_runtime/rpc/grpc_server_lib.cc:365] Started server with target: grpc://localhost:34146
2020-08-21 12:50:30,159:INFO:tf_yarn._task_commons: Starting execution chief:0
2020-08-21 12:50:30,184:INFO:tensorflow: Not using Distribute Coordinator.
2020-08-21 12:50:30,214:WARNING:tensorflow: From /appdata/hdfs/v3/yarn/nm/usercache/gpachitariu/appcache/application_1588085646385_2459521/container_e160_1588085646385_2459521_01_000004/pyenv/lib/python3.6/site-packages/tensorflow_core/python/ops/resource_variable_ops.py:1630: calling BaseResourceVariable.__init__ (from tensorflow.python.ops.resource_variable_ops) with constraint is deprecated and will be removed in a future version.
Instructions for updating:
If using Keras pass *_constraint arguments to layers.
2020-08-21 12:50:30,215:WARNING:tensorflow: From /appdata/hdfs/v3/yarn/nm/usercache/gpachitariu/appcache/application_1588085646385_2459521/container_e160_1588085646385_2459521_01_000004/pyenv/lib/python3.6/site-packages/tensorflow_core/python/training/training_util.py:236: Variable.initialized_value (from tensorflow.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Use Variable.read_value. Variables in 2.X are initialized automatically both in eager and graph (inside tf.defun) contexts.
2020-08-21 12:50:30,648:INFO:tensorflow: Calling model_fn.
2020-08-21 12:50:30,711:WARNING:tensorflow: From /appdata/hdfs/v3/yarn/nm/usercache/gpachitariu/appcache/application_1588085646385_2459521/container_e160_1588085646385_2459521_01_000004/pyenv/lib/python3.6/site-packages/tensorflow_core/python/feature_column/feature_column_v2.py:518: Layer.add_variable (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.
Instructions for updating:
Please use `layer.add_weight` method instead.
2020-08-21 12:50:30,718:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='alcohol', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,727:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='chlorides', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,737:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='citric_acid', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,746:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='density', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,755:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='fixed_acidity', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,764:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='free_sulfur_dioxide', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,774:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='pH', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,783:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='residual_sugar', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,793:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='sulphates', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,802:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='total_sulfur_dioxide', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,813:DEBUG:tensorflow: Transforming feature_column NumericColumn(key='volatile_acidity', shape=(1,), default_value=None, dtype=tf.float32, normalizer_fn=None).
2020-08-21 12:50:30,865:WARNING:tensorflow: From /appdata/hdfs/v3/yarn/nm/usercache/gpachitariu/appcache/application_1588085646385_2459521/container_e160_1588085646385_2459521_01_000004/pyenv/lib/python3.6/site-packages/tensorflow_estimator/python/estimator/canned/linear.py:308: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.
2020-08-21 12:50:31,338:WARNING:tensorflow: From /appdata/hdfs/v3/yarn/nm/usercache/gpachitariu/appcache/application_1588085646385_2459521/container_e160_1588085646385_2459521_01_000004/pyenv/lib/python3.6/site-packages/tensorflow_core/python/keras/optimizer_v2/ftrl.py:143: calling Constant.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.
Instructions for updating:
Call initializer instance with the dtype argument instead of passing it to the constructor
2020-08-21 12:50:31,484:INFO:tensorflow: Done calling model_fn.
2020-08-21 12:50:31,484:INFO:tensorflow: Create CheckpointSaverHook.
